cmd/mattermost/commands/config.go

```diff
@@ -320,6 +328,109 @@ func UpdateMap(configMap map[string]interface{}, configSettings []string, newVal
 	}
 }
 
+func configResetCmdF(command *cobra.Command, args []string) error {
+	app, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+
+	defer app.Shutdown()
+
+	currentConfig := app.Config()
+	defaultConfig := &model.Config{}
+	defaultConfig.SetDefaults()
+
+	if len(args) == 0 {
+		var confirmResetAll string
+		CommandPrettyPrintln("Are you sure you want to reset all the configuration settings?(YES/NO): ")
+		fmt.Scanln(&confirmResetAll)
+		if confirmResetAll == "YES" {
+			currentConfigMap := configToMap(*currentConfig)
```

Comment:
This option should be atomic and simpler. The `app.UpdateConfig` is already storing the configuration in each iteration, so, if something goes wrong in one iteration, you will end up with a partial reset of the configuration.

Here probably you simply need something like `app.SaveConfig(defaultConfig, false)`.